posydon/popsyn/synthetic_population.py

```diff
@@ -556,6 +602,35 @@ def get_dco_detection_rate(self, sensitivity='design_H1L1V1', export_cols=None,
         # TODO: store p_det
         self.df_detectable = self.resample_synthetic_population(index, z_formation, z_merger, w_ijk, export_cols=export_cols)
 
+    def get_formation_channels(self):
+        """Get formation channel and add to df and df_online."""
+        
+        # loop through each binary
+        unique_binary_index = np.unique(self.df.index)
+        for index in unique_binary_index:
+
+            # get event column and information from interpolated classes
+            df_binary = self.df.loc[index,['event']].dropna()
+            df_binary_online = self.df_oneline.loc[index,['interp_class_HMS_HMS','interp_class_CO_HMS_RLO', 'interp_class_CO_HeMS']].dropna()
```

Comment:
Consider to not hard code the interpolation classes, e.g. first generate a list with all which contain "interp_class".